server/chat/backend/agent/providers/openrouter_provider.py

```
@@ -97,5 +103,38 @@ def get_native_model_name(self, model: str) -> str:
+            )
+            if resp.status_code == 200:
+                data = resp.json().get("data", [])
+                # OpenRouter ids are already in "provider/model" form.
+                live = [m["id"] for m in data if m.get("id")]
+                if live:
+                    self._models_cache = live
+                    self._models_cache_time = now
+                    return live
+            else:
+                logger.warning("OpenRouter /models returned %s", resp.status_code)
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)